src/server/src/main/java/io/cassandrareaper/resources/RepairRunResource.java

```diff
@@ -320,6 +311,26 @@ static Response checkRequestForAddRepair(
     return null;
   }
 
+  @Nullable
+  private static Response validateIntensity(Optional<String> intensityStr) {
+    if (intensityStr.isPresent()) {
+      try {
+        final Double intensity = Double.parseDouble(intensityStr.get());
+        if (intensity <= 0.0 || intensity > 1.0) {
+          return Response.status(Response.Status.BAD_REQUEST)
+              .entity("query parameter \"intensity\" must be in half closed range (0.0, 1.0]: " + intensityStr.get())
+              .build();
+        }
+      } catch (NumberFormatException ex) {
+        LOG.error(ex.getMessage(), ex);
+        return Response.status(Response.Status.BAD_REQUEST)
+            .entity("invalid value for query parameter \"intensity\": " + intensityStr.get())
+            .build();
+      }
+    }
+    return null;
+  }
```

Comment:
this is trivial… but what about 
```
   private static double parseIntensity(Optional<String> intensityStr) throws ValidationException {
       Preconditions.checkArgument(intensityStr.isPresent());
       try {
         final Double intensity = Double.parseDouble(intensityStr.get());
         if (intensity <= 0.0 || intensity > 1.0) {
           throw new ValidationException("query parameter \"intensity\" must be in half closed range (0.0, 1.0]: " + intensityStr.get());
         }
       } catch (NumberFormatException ex) {
         throw new ValidationException("invalid value for query parameter \"intensity\": " + intensityStr.get());
       }
       return Double.parseDouble(intensityStr.get());
   }
```

then the lines below could be…
```
    if (intensityStr.isPresent()) {
       try {
           return updateRepairRunIntensity(repairRun.get(), repairUnit.get(), segmentsRepaired, parseIntensity(intensityStr));
        } catch (ValidationException ex) {
           return Response.status(Response.Status.BAD_REQUEST).entity(ex.getMessage()).build();
        }
     }
```
(and the previous and first call to `validateIntensity(..)` from `checkRequestForAddRepair(..)` would remain inside the `if (intensityStr.isPresent()) {..}` conditional)

Reply:
It will be exception logic. I prefer to avoid that in case when I know what can be wrong. Exceptions for exceptional situations. It's better to implement DDD conceptions here.
Intensity — it's a value object. It's not double actually, because for this domain logic it's float value between (0.0, 1.0]. So better to separate domain logic and infrastructure part of code (like working with db layer and working with api layer). But there is a lot of work for refactoring.
1. First of all RepairRun object must have field intensity with it own type like `private Intensity intensity`.
2. Domain data type Intensity must contain logic about it rightness (like value boundaries (0.0, 1.0]).
3. Finally project must provide opportunity for correct object creation (through the factory or static factory method).

I will encapsulate this idea only on RepairRunresource level, because have no time for huge refactoring.

Reply:
> Exceptions for exceptional situations. 

I would agree with this over public methods and public APIs, but here we're working privately inside the class.

The approach I suggested was to minimise the patch size: less code (so long as it is separates concerns and is readable) means easier evolution and less bugs. The addition of Intensity and Result classes outside of the package unnecessarily expands the API surface. 

I'll take what you got that is good and code up what I was thinking…

Reply:
> I'll take what you got that is good and code up what I was thinking…

This is more along the lines of what I was thinking:
https://github.com/thelastpickle/cassandra-reaper/commit/e3e772945f1b8676140a74a944fbb3fc1a33e964

this provides the feature in half the patch size.

Reply:
> Result classes outside of the package unnecessarily expands the API surface.
What are you talking about? Result and Intensity encapsulated at cassandra-reaper-server. I used them for RepairRunResource only.

Reply:
> What are you talking about? Result and Intensity encapsulated at cassandra-reaper-server. I used them for RepairRunResource only.

 By "package" here I meant the java package. Result and Intensity had been put into `io.cassandrareaper.core`, a package only for model beans.
The view beans are in a different package `io.cassandrareaper.resources.view` (which i don't agree with, they should be in the resources package).
And we have no validation (or input) beans yet.